content/articles/phone-verification-in-laravel-using-twilio/index.md

```diff
@@ -0,0 +1,1236 @@
+    {
+
+        return view('auth.verification');
+
+    }  
+
+    public function registration()
+
+    {
+
+        return view('auth.registration');
+
+    }
+    public function postRegistration(Request $request)
+
+    { 
+
+        $request->validate([
+
+            'name' => 'required',
+
+            'phone_number' => 'required|numeric|unique:users',
+
+            'password' => 'required|min:8',
+
+        ]);
+
+        $code = random_int(100000, 999999);
+
+        $user = new User;
+
+        User::create([
+
+            'name' => $request->name,
+
+            'phone_number' => $request->phone_number,
+
+            'verification_code' => $code,
+
+            'password' => Hash::make($request->password)
+
+          ]);
+    
+        return redirect()->route('login')->withSuccess('login successfull');
+
+    } 
+
+ public function postLogin(Request $request)
+
+    {
+
+        $request->validate([
+
+            'phone_number' => 'required',
+
+            'password' => 'required',            
+
+        ]);
+        
+        $user = User::where('phone_number', $request->phone_number)->first();
+
+        if($user && $user->isVerified == false && Hash::check($request->password,     $user->password)){
+
+            return redirect("login")->with('status', 'Account not verified');
+
+        };
+        $credentials = $request->only('phone_number', 'password');       
+
+        if (Auth::attempt($credentials)) {           
+
+            return redirect()->intended('dashboard')
+
+                        ->withSuccess('You have Successfully loggedin');
+        }  
+
+        return redirect("login")->witherrors('Oppes! You have entered invalid credentials');
+
+    }
+
+    public function dashboard()
+
+    {
+
+        if(Auth::check()){
+
+            return view('dashboard');
+
+        }
+
+        return redirect("login")->witherrors('Opps! You do not have access');
+
+    }
+    public function logout() {
+
+        Session::flush();
+
+        Auth::logout();  
+
+        return Redirect('login');
+
+    }
+
+    }
+```
+We have now added all that we need to authenticate a user in our `app/Http/Controllers/Auth/AuthController.php`. The `index `method, the `registration` method, and the `reverify` method returns the `login`, `registration`, and `verification` view respectively. The `dashboard` method returns the `dashboard` view only if the user is authenticated, else it will redirect to the `login` with an error message. This is done using the `Auth::check()`. The `postRegistration` method saves all the user’s input and makes sure that the data provided by the user passes the validation check, it then redirects the user to the login page with a success message. The `postLogin` method authenticates the user base on the credentials provided by the user and the verification status. this uses the `Auth::attempt($credentials)`. The `logout` method logs the user out and also clears the session using the `Session::flush()`
```

Comment:
Try breaking all these down, for example, explaining each method on its own line?